tools/aux/tpm2_print.c

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2017, National Instruments
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <stdio.h>
+#include "log.h"
+#include "files.h"
+#include "tpm2_util.h"
+#include "tpm2_alg_util.h"
+
+#define FILE_2_SHORT_ERROR(filename) LOG_ERR("%s: File too short", (filename))
+
+static void print_hex_buffer(const UINT8* buff, UINT32 size) {
+    while(size-- > 0) {
+        printf("%2.2x", (unsigned int)*buff);
+        ++buff;
+    }
+    printf("\n");
+}
+
+static bool print_hex(FILE* fd, const char* const filename, UINT32 size) {
+    UINT8 byte;
+    while(size-- > 0) {
+        if(fread(&byte, 1, 1, fd) != 1) {
+            FILE_2_SHORT_ERROR(filename);
+            printf("\n");
+            return false;
+        }
+        printf("%2.2x", (unsigned int)byte);
+    }
+    printf("\n");
+    return true;
+}
+
+static bool print_tpm2b_hex(FILE* fd, const char* const filename) {
+    UINT16 size;
+    if(!files_read_16(fd, &size)) {
+        FILE_2_SHORT_ERROR(filename);
+        return false;
+    }
+    return print_hex(fd, filename, size);
+}
+
+static bool print_clock_info(FILE* fd, const char* const filename, const char* const prefix) {
+    union {
+        UINT8  u8;
+        UINT32 u32;
+        UINT64 u64;
+    } numb;
+
+    if(!files_read_64(fd, &numb.u64)) {
+        goto read_error;
+    }
+    printf("%s.clockInfo.clock=%llu\n", prefix, (long long unsigned int)numb.u64);
```

Comment:
Everything should be output in yaml format, for example, see #618 
Also use tpm2_tool_output() so it respects the -Q option.

Reply:
Done